### DISCUSSION TITLE
縦一列のキーが反応しない(keyball39)

### DISCUSSION THREAD

**Discussion by kenswin-oga** [Yowkees/keyball#456] [Q&A]
電子工作初心者です。<br>
特定のキーが反応しないため、解決方法をご教授いただけないでしょうか。<br><br>

現在、組み立てマニュアルにおいて「3-7ProMicroの書き込みと動作確認」が終わった段階です。<br>
ピンセットでスイッチのパッドをショートさせて動作確認したところ、写真の赤丸で示すように、トラックボール側基盤のpromicro側のスイッチ5つが反応しません。<br>
![IMG20231226002235](https://github.com/Yowkees/keyball/assets/131173350/a1c0023f-c8d7-4f3b-8475-be4bb7d65344)
![IMG20231226002119](https://github.com/Yowkees/keyball/assets/131173350/21153655-1b3b-4571-a3ab-8f84d8951af8)

縦一列が反応しないので、ソケットやダイオードではなくpromicro側の問題なのかなと思ってます。<br>
確認したことは以下です。<br>
- ピンセットで該当のスイッチのパッドをショートさせ→入力できない
- 該当のソケットと対応するダイオードが導通しているか→していた

promicroに繋がっているかの確認方法や、そのほか確認すべき点などご教授いただけましたら幸いです。
お手数おかけしますがよろしくお願いします。

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

kenswin-oga様のおっしゃる通り、縦一列の反応が無い場合はProMicro周辺に原因がある可能性が高いです。
今回の反応のないキーの列に対応する端子はCOL４、COL５の端子になりますのではんだ付けの見直しをお願いいたします。

**Comment by kenswin-oga** [comment]
素早い対応ありがとうございます！<br>
ご指摘通りcol4, col5のはんだ付けを目視で確認しましたが、問題ないように見えますがどうでしょうか？<br>
また、その確認方法などございますでしょうか？<br>
![IMG20231226143701](https://github.com/Yowkees/keyball/assets/131173350/546c7080-751c-4daf-80ad-37fbb3077ab4)
<br>
加えて、pinmicro側で以下写真の赤丸の端子の基盤が少し死んでいるように見えたのですが、これが原因になり得ますでしょうか？<br>
質問ばかりで申し訳ございませんがよろしくお願いします。
![IMG20231226144006](https://github.com/Yowkees/keyball/assets/131173350/a198b0f8-f1cc-48f3-8601-d6453da02030)

**Comment by Ryo624** [comment]
ご確認していただきましてありがとうございます。

こちらの伝え方が悪く申し訳ありません。該当箇所のはんだ付けをやり直していただけますと幸いです。
ProMicroの赤丸部分も新しいはんだを加えながらはんだ付けをやり直していただいて状況を再度ご連絡ください。

**Comment by kenswin-oga** [comment]
ありがとうございます。<br>
ご指摘通りCOL４、COL５の端子をはんだ付けしなおしましたが、状況は変わらずでした。。。

**Comment by kenswin-oga** [comment]
12pinソケットを別のものに付け替えてもみましたが、効果なしでした。

**Comment by Ryo624** [comment]
はんだ付け等のやり直しを実施していただきましてありがとうございます。

写真で示しています緑点がCOL４、赤点がCOL５となっていますので、それぞれが導通しているかを確認していただいてもよろしいでしょうか。

こちらで導通が確認されても不具合が続く場合はProMicro自体に原因があるかもしれませんので、左右のProMicroを入れ替えて再度症状の確認をお願いいたします。

![292767678-a1c0023f-c8d7-4f3b-8475-be4bb7d65344](https://github.com/Yowkees/keyball/assets/143698349/4bb1473f-5eb6-4e06-bd67-0b24c0671cce)

**Comment by Ryo624** [comment]
重ねてのご連絡失礼します。

添付しました写真の表側のProMicroの緑点と赤点もCOL４、COL5の端子となっている場所ですので、こちらも隣同士の端子がショートしていないかを核にしながらはんだ付けをやり直していただけますと幸いです。
![292767685-21153655-1b3b-4571-a3ab-8f84d8951af8](https://github.com/Yowkees/keyball/assets/143698349/ca9ae3ed-f8bd-48b9-8509-e39426ea05e6)

また、先ほど添付しました導線を確認していただく緑点と赤点ですがProMicro側の右側の緑点と赤点2点は無視してください。

**Comment by kenswin-oga** [comment]
少し時間がたっての返信恐れ入ります。

一枚目で示していただいた箇所の導通を確認しましたが、どれも導通できていませんでした。

また、ついでに他のキースイッチとcolの組み合わせも確認しましたが、どれも導通できていませんでした。
(ピンセットでのショートによる入力はできているものです。もう片方の基盤も同様でした)
このようなことは起き得るのでしょうか。。

導通の確認方法が悪いのかとも思いましたが、キースイッチの端子同士やキースイッチとダイオードの導通は確認できております。
なにか分かることがありましたらご教授いただきたいです。
よろしくお願いいたします。

**Comment by Ryo624** [comment]
導通の確認を実施していただきましてありがとうございます。

すでの試したうえでのご連絡であれば大変申し訳ありませんが、左右のProMicroを入れ替えても同様の症状は起きますでしょうか。
お手数ですがご確認をよろしくお願いいたします。

**Comment by kenswin-oga** [comment]
確認いたしました。<br>
左右のpromicroを入れ替えても、同じキースイッチのみ入力できませんでした。(ピンセットによるショートでの入力)

**Comment by Ryo624** [comment]
ご対応ありがとうございます。

修正後の基板を確認できていませんが最初に添付していただいた写真のProMicro部分に端子がショートしているように見える箇所がありますので、赤点が示した部分のようなはんだが現在も同様に確認できる場合ははんだ付けをやり直していただいてもよろしいでしょうか。
![292811739-546c7080-751c-4daf-80ad-37fbb3077ab4](https://github.com/Yowkees/keyball/assets/143698349/42a0eb9a-9666-4a73-9805-b2e531eee2bd)

改善が見られない場合、ProMicroとCOL部分の導通が確認できないこと、ProMicroを入れ替えても症状が変わらないということから、おそらく何かの拍子に基板の配線が断線してしまったと考えられます。
白銀ラボのウェブショップから基板と部品セットの販売が可能ですので、お問い合わせフォームからご連絡いただけますでしょうか。
https://shirogane-lab.net/inquiry
